Cognoma.py

```diff
@@ -0,0 +1,662 @@
+                            InducedSNPpresentToExp[:, x]))
+            AllSyntheticEffect = np.append(
+                AllSyntheticEffect, regressResults.slope)
+            AllSyntheticPVal = np.append(
+                AllSyntheticPVal, regressResults.pvalue)
+
+        # Make a scatter plot of the difference between three of the different
+        # expression
+        # Shows the differnece between the transformed domains and the actual
+        # difference in the real data.
+        plt.figure()
+        plt.title("Scatter Plot of the differentially expressed genes between"
+                  "real TP53 wildtype vs. real TP53 mutated and real vs."
+                  " synthetic TP53 data")
+        plt.scatter(RealEffect, SyntheticEffect, c='r', alpha=0.03,
+                    label="Real TP53 Wildtype vs. Synthetic TP53 Mutated")
+        plt.scatter(RealEffect, SyntheticNotEffect, c='b', alpha=0.03,
+                    label="Real TP53 Mutated vs. Synthetic TP53 Wildtype")
+        plt.xlabel("Effect of real TP53 Wildtype vs. real mutated")
+        plt.ylabel("Effect of real and synthetic data")
+        lgd = plt.legend(ncol=1,
+                         bbox_to_anchor=(1.03, 1),
+                         loc=2,
+                         borderaxespad=0.,
+                         fontsize=10)
+        plt.savefig(os.path.join('Output', params.dataSetInfo.name,
+                                 'Scatter_{}.png'.
+                                 format(str(params.outputNum))),
+                    bbox_extra_artists=(lgd,),
+                    bbox_inches='tight')
+
+        #######################################################################
+        # For creating bar graphs of latent space expression between
+        # TP53 Wiltype and TP53 mutated
+
+        # Find the difference in the nodes between TP53 wildtyupe and mutated
+        # in the right model
+        SNPpresentLatent = modelHandle.rightEncoder.predict(SNPpresent)
+        SNPabsentLatent = modelHandle.rightEncoder.predict(SNPabsent)
+
+        # Turn this into a percentage difference
+        LantentSpacePerc = (abs(SNPpresentLatent - SNPabsentLatent)
+                            / abs((SNPabsentLatent + SNPabsentLatent)/2))
+
+        LantentSpacePerc = LantentSpacePerc.mean(axis=0)
+
+        # Create a bar graph showing which nodes are most different
+        plt.figure()
+        plt.bar(np.arange(LantentSpacePerc.shape[0]), LantentSpacePerc)
+        title = plt.title("Percentage difference between TP53 Wildtype and"
+                          " mutated for each latent space node, Mutation")
+        plt.xticks(np.arange(LantentSpacePerc.shape[0]), np.arange(
+            LantentSpacePerc.shape[0]))
+        plt.xlabel("Latent Space Nodes")
+        plt.ylabel("Latent Space Percentage Difference")
+        plt.savefig(os.path.join('Output', params.dataSetInfo.name,
+                                 'TP53MutLatentSpaceComparrison_{}.png'.
```

Comment:
If you're using Python 3.6+, could do 
```python
f'TP53MutLatentSpaceComparrison_{params.outputNum}.png'
```

Reply:
I do need to switch to Python 3.6, there's no reason that I'm using 2.7 other than familiarity.